src/lists.rs

```diff
@@ -41,11 +47,38 @@ pub struct ListFormatting<'a> {
     pub h_width: usize,
     // Available width if we layout vertically
     pub v_width: usize,
+    // Non-expressions, e.g. items, will have a new line at the end of the list.
+    // Important for comment styles.
+    pub is_expression: bool
```

Comment:
I agree it's not pretty, but I am afraid it will be a bit irregular in any case. If I understand correctly, we have to decide between 
```rust
(item1, // Comment 1
 item2, // Comment 2
 item3, /* Comment 3 */)
```
and
```rust
(item1, // Comment 1
 item2, // Comment 2
 item3, // Comment 3
)
```
Personally, I don't believe the latter is always better than the first, but I can get behind it if that's what is decided on.